### PR TITLE
BREAKING: web: change Error interface, improve HEAD support of HTTPError and add AsError()/AsErrorWithCode() helpers

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

**Review by karasz — CHANGES_REQUESTED** [review]
see comment

File: web/errors.go [hunk 1]
```diff
@@ -27,8 +27,8 @@ type HTTPError struct {
 	Hdr  http.Header
 }
 
-// Status returns the StatusCode associated with the Error
-func (err *HTTPError) Status() int {
+// HTTPStatus returns the StatusCode associated with the Error
```
[the remark this answers is not in the document]
Comment (amery) [reply web/errors.go:30]: Oops

**Comment by amery** [comment]
@karasz PR amended as recommended

**Review by karasz — APPROVED** [review]
LGTMN

**Comment by amery** [comment]
`darvaza.org/x/web@v0.5.0` tagged

**Pull request merged**